oak/server/rust/oak_runtime/src/lib.rs

```diff
@@ -1,645 +1,152 @@
-    }
-    pub fn write_message(&mut self, msg: OakMessage) -> u32 {
-        if self.direction != Direction::Write {
-            return OakStatus::ERR_BAD_HANDLE.value() as u32;
-        }
-        self.channel
-            .write()
-            .expect("corrupt channel ref")
-            .write_message(msg)
-    }
-}
+use node::NodeConfiguration;
+use oak_abi::OakStatus;
+use proto::manager::NodeConfiguration_oneof_config_type;
 
-impl Clone for ChannelHalf {
-    fn clone(&self) -> ChannelHalf {
-        ChannelHalf::new(self.direction, self.channel.clone())
-    }
-}
+/// TODO: Items here are not provided by `core` or `alloc`. These need to be provided
+/// by an external dependency i.e. asylo_rust when targeting asylo.
+#[cfg(feature = "std")]
+mod platform {
+    pub type HashMap<K, V> = std::collections::HashMap<K, V>;
+    pub type RwLock<T> = std::sync::RwLock<T>;
+    pub type Mutex<T> = std::sync::Mutex<T>;
+    pub use std::thread::current; // current().id()
+    pub use std::thread::spawn;
+    pub type JoinHandle = std::thread::JoinHandle<()>;
 
-impl Drop for ChannelHalf {
-    fn drop(&mut self) {
-        // Update the referenced channel's counts of extant halves.
-        match self
-            .channel
-            .write()
-            .expect("corrupt channel ref")
-            .half_count
-            .entry(self.direction)
-        {
-            Entry::Occupied(mut e) => {
-                if *e.get() <= 0 {
-                    panic!("negative channel ref count");
-                }
-                *e.get_mut() -= 1;
-            }
-            Entry::Vacant(_) => panic!("missing enum value in half_count"),
-        };
+    pub fn yield_thread() {
+        std::thread::sleep(std::time::Duration::from_millis(10));
     }
 }
 
-#[derive(PartialEq, Clone, Debug, Eq, Hash)]
-pub struct WasmEntrypointFullName {
-    // Name of the config stanza for the Node.
-    pub config_name: String,
-    // Entrypoint name.
-    pub entrypoint_name: String,
-}
+use platform::*;
 
-pub struct OakRuntime {
-    termination_pending: bool,
-    node_configs: HashMap<String, NodeConfiguration_oneof_config_type>,
-    // Map entrypoint names to a test entrypoint function pointer.
-    entrypoints: HashMap<WasmEntrypointFullName, NodeMain>,
-    // Map name of Node config names to next index.
-    next_index: HashMap<String, u32>,
-    // Track a reference to the write half of the channel used for sending
-    // gRPC method notifications in to the Application under test.
-    grpc_in_half: Option<ChannelHalf>,
-    // Node instances organized by internal Node name.
-    nodes: HashMap<String, OakNode>,
+/// TODO: doc
+pub struct OakMessage {
+    pub data: Vec<u8>,
+    pub channels: Vec<ChannelEither>,
 }
 
-struct OakNode {
-    halves: HashMap<Handle, ChannelHalf>,
-    // Handle for a thread running the main loop for this node.
-    thread_handle: Option<std::thread::JoinHandle<()>>,
+/// TODO: doc
```

Comment:
This `TODO` would be nice to have done before the PR is merged &ndash; the previous runtime code had a bunch more doc comments (cf. eg. 0ea930a8009ca2f1457a272df6adb4ad053c0a53).